src/borg/crypto.pyx

```diff
@@ -6,6 +6,15 @@ from cpython.buffer cimport PyBUF_SIMPLE, PyObject_GetBuffer, PyBuffer_Release
 API_VERSION = 3
 
 
+cdef extern from "blake2/blake2b-ref.c":
+    ctypedef struct blake2b_state:
+        pass
+
+    int blake2b_init(blake2b_state *S, size_t outlen) nogil
+    int blake2b_update(blake2b_state *S, const void *input, size_t inlen) nogil
+    int blake2b_final(blake2b_state *S, void *out, size_t outlen) nogil
```

Comment:
Yes, a compatible implementation can be written with anything that gives BLAKE2b, no special features (keying, tree hashing, personalization etc.) needed. This makes it compatible with the OpenSSL API (L. 237). Python 3.6 hashlib has the full API and is also compatible


However, to be very sure about that, I'll produce a small test program against OpenSSL 1.1